From 586e4d179b9af1d8fb95405d4ec25cb16a189abd Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert+renesas@glider.be>
Date: Wed, 26 Aug 2015 19:21:00 +0200
Subject: [PATCH] kbuild: Enable DT symbols when CONFIG_OF_OVERLAY is used

This patch is taken from [1], topic/overlays branch.
U-Boot overlay feature requires the base dtb to be built with symbols.
So build the dtbs with symbols when overlay support (CONFIG_OF_OVERLAY) is
enabled.

[1] https://git.kernel.org/pub/scm/linux/kernel/git/geert/renesas-drivers.git

Change-Id: Ia98091131777ac663f3fae9093e474e02e9ed4a7
Signed-off-by: Geert Uytterhoeven <geert+renesas@glider.be>
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
---
Replacement for "ARM: DT: Enable symbols when CONFIG_OF_OVERLAY is
used".
---
 scripts/Makefile.lib | 4 ++++
 1 file changed, 4 insertions(+)

--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -388,6 +388,10 @@ $(obj)/%.dtb.S: $(obj)/%.dtb FORCE
 $(obj)/%.dtbo.S: $(obj)/%.dtbo FORCE
 	$(call if_changed,wrap_S_dtb)
 
+ifeq ($(CONFIG_OF_OVERLAY),y)
+DTC_FLAGS += -@
+endif
+
 quiet_cmd_dtc = DTC     $@
 cmd_dtc = $(HOSTCC) -E $(dtc_cpp_flags) -x assembler-with-cpp -o $(dtc-tmp) $< ; \
 	$(DTC) -o $@ -b 0 \
